ZUGFeRD/InvoiceDescriptor.cs

```diff
@@ -812,9 +899,14 @@ public void SetContractReferencedDocument(string contractNo, DateTime? contractD
         ///
         /// You might use AddTradeAllowanceCharge() instead.
         /// </summary>
+        /// <param name="amount">Charge amount</param>
+        /// <param name="description">Description of the charge</param>
+        /// <param name="taxTypeCode">Type of tax</param>
+        /// <param name="taxCategoryCode">Tax category</param>
+        /// <param name="taxPercent">Tax percentage</param>
         public void AddLogisticsServiceCharge(decimal amount, string description, TaxTypes taxTypeCode, TaxCategoryCodes taxCategoryCode, decimal taxPercent)
         {
-            this.ServiceCharges.Add(new ServiceCharge()
+            GetLogisticsServiceCharges()?.Add(new ServiceCharge()
```

Comment:
can you please revert the Get*()?.Add() calls?
The obsolete warnings are just temporary